hs_core/templatetags/hydroshare_tags.py

```diff
@@ -116,13 +116,12 @@ def resource_first_author(content):
             first_creator = creator
             break
 
-    if first_creator:
-        if first_creator.name and first_creator.relative_uri and first_creator.is_active_user:
-            return format_html('<a href="{desc}">{name}</a>',
-                               desc=first_creator.relative_uri,
-                               name=first_creator.name)
-        elif first_creator.name:
-            return format_html('<span>{name}</span>', name=first_creator.name)
+    if first_creator.name and first_creator.relative_uri and first_creator.is_active_user:
+        return format_html('<a href="{desc}">{name}</a>',
+                            desc=first_creator.relative_uri,
+                            name=first_creator.name)
+    elif first_creator.name:
+        return format_html('<span>{name}</span>', name=first_creator.name)
     else:
         first_creator = content.metadata.creators.filter(order=1).first()
```

Comment:
Thanks @pkdash you're right but I think we did that for loop in order to avoid a query on every resource on the My Resources page
https://github.com/hydroshare/hydroshare/commit/de89b52b875224769034da4418f5947f65b0057c

Am I right that if we just use 
`first_creator = content.first_creator`
instead of 
`first_creator = content.metadata.creators.filter(order=1).first()`
^^^still, both of these will hit the database for _every single resource in the list_?

Reply:
Yeah you are right. Now I remember I probably wrote that `for` loop to avoid hitting the db for each resource in my resources page. I forgot that we are fetching `creators` part of the query for the my resources page. Sorry for my confusion. In that case I would just remove the following line that is part of the `if else` block
`first_creator = content.metadata.creators.filter(order=1).first()`

Reply:
sounds good, I made that change